scripts/travis-run.sh

```diff
@@ -34,22 +34,32 @@ then
 
     # If the environment vars changed (e.g., boost, R, perl) then there's no
     # good way of knowing which recipes need rebuilding so we check them all.
-    #
-    # For cron jobs we always want to check everything.
     set +e
     git diff --exit-code --name-only $RANGE scripts/env_matrix.yml
     ENV_CHANGE=$?
     set -e
-    if [ $ENV_CHANGE -eq 1 ] || [ $TRAVIS_EVENT_TYPE = "cron" ]
+
+    if [[ $ENV_CHANGE -eq 1 && $TRAVIS_BRANCH == "master" && $TRAVIS_PULL_REQUEST != "false" ]]
+    then
+        echo ""
+	echo "Env matrix was modified but this pull request does not target a bulk/* branch."
+	echo "Please target a branch named bulk/<reasonable-name>. Rebuild and PR against that branch until all builds succeed."
```

Comment:
If Travis times out because there are too many packages to be build, you can restart the build job until it has finally processed all recipes.